src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1602,12 +1420,12 @@ rpmostree_sysroot_upgrader_deploy (RpmOstreeSysrootUpgrader *self,
   const char *overlay_v[] = { NULL, NULL };
   if (g_hash_table_size (rpmostree_origin_get_initramfs_etc_files (self->origin)) > 0)
     {
-      g_auto(GLnxTmpfile) tmpf = { 0, };
-      if (!generate_initramfs_overlay (rpmostree_origin_get_initramfs_etc_files (self->origin),
-                                       &tmpf, cancellable, error))
+      glnx_fd_close int fd = -1;
+      if (!ror_initramfs_overlay_generate (rpmostree_origin_get_initramfs_etc_files (self->origin),
+                                           &fd, error))
```

Comment:
I'm a bit sad to see the `cancellable` gone here, especially because this ROR replacement is both IO-intensive and with unbounded recursion.

Any hope we could bridge `GCancellable` back and forth across FFI border, and use it to kill at least unbounded looping/recursive logic? I saw that `gio-sys` has something, but it's not clear to me how to properly use that.

Reply:
What's unbounded?  The filesystem can't be unbounded, and I guess in theory sure `cpio` or `gzip` could have an infinite loop bug but...

It would be good to honor the cancellable; but in the end what we really need to do fork off transactions as a subprocess and just kill them to cancel; it'd be way more reliable and also allow us to sandbox more.

Reply:
I was referring to `list_files_recurse`, which needs to exhaust the whole input space in order to return, and used to be cancellable before. However I did not really want to focus on the specific function, but on generally trying to gauge how you were planning to tackle cancellation in the new ROR world.

Reply:
> I was referring to list_files_recurse, which needs to exhaust the whole input space in order to return, and used to be cancellable before. 

True...but the fact that we require the input filename to start with `/etc` greatly narrows the possible footguns here; you can't `rpm-ostree ex initramfs --track /`.  

Anyways I proxied through the cancellable.